### PR TITLE
Remove runtime dependency from solana-transaction-status

### LINKED ISSUE #26387: solana-transaction-status: factor out solana-runtime dependency
#### Problem

`solana_transaction_status` uses `solana_runtime::bank::RewardType` and `solana_runtime` introduces a lot of unnecessary dependencies to `solana_transaction_status`, some of which make it less portable.

#### Proposed Solution

Move the `RewardType` enum somewhere else so it doesn't bring so many dependencies with it.

### PR DESCRIPTION
**Pull request by CriesofCarrots**
#### Problem
The dependency on `solana-runtime` introduces a lot of unnecessary dependencies to `solana-transaction-status`, some of which make the crate less portable.

#### Summary of Changes
- Move RewardType to `solana-sdk`, but re-export it from runtime to not break current callers
- Move `collect_token_balances()` to `solana-ledger`, but balance structs stay in `solana-transaction-status`

Fixes #26387

### CODE REVIEW

**Comment by CriesofCarrots** [comment]
To be explicit about lock changes:

- solana-ledger: spl-token and spl-token-2022 were already in the dependency tree via solana-transaction-status; just called out directly now
- `programs/bpf/rust`: needs a dev dependency on solana-ledger to use the `collect_token_balances()` method. If this is unacceptable, we'll either need to move that method into its own crate, or move the affected tests elsewhere

**Comment by CriesofCarrots** [comment]
@jackcmay , could you please look at the `programs/bpf/rust` change, or recommend another reviewer?

File: runtime/src/bank.rs [hunk 1]
```diff
@@ -35,6 +35,7 @@
 //! already been signed and verified.
 #[allow(deprecated)]
 use solana_sdk::recent_blockhashes_account;
+pub use solana_sdk::reward_type::RewardType;
```
Comment (joncinque) [inline runtime/src/bank.rs:38]: nit: what do you think about marking this re-export as deprecated? That'll just require changing the other places that import this

**Review by joncinque — APPROVED** [review]
Looks great! I came up with a much worse solution.  Adding the ledger dependency to dev-dependencies looks fine to me, but I'll let someone else officially approve that part.

File: runtime/src/bank.rs (hunk 1 above)
Comment (CriesofCarrots) [reply runtime/src/bank.rs:38]: Ah, that's a nice idea. The other monorepo importers can just be updated to point to sdk.

**Review by jackcmay — APPROVED** [review]

File: runtime/src/bank.rs (hunk 1 above)
Comment (CriesofCarrots) [reply runtime/src/bank.rs:38]: Sadly, it turns out a deprecation tag here won't actually raise a warning for users: https://github.com/rust-lang/rust/issues/30827

Comment (joncinque) [reply runtime/src/bank.rs:38]: Ah, that's too bad.  Thanks for looking into it

**Pull request merged**